lib/Runtime/Library/JavascriptArray.inl

```diff
@@ -1357,6 +1397,8 @@ SECOND_PASS:
                     const bool currentWasFull = current->length == current->size;
 
                     Assert(itemIndex == current->left + current->size);
+                    SparseArraySegmentBase* oldSegment = current;
+                    bool isInlineSegment = JavascriptArray::IsInlineSegment(oldSegment, this);
                     current = SparseArraySegment<T>::CopySegment(recycler, (SparseArraySegment<T>*)current, next->left, next, next->left, next->length);
```

Comment:
>SparseArraySegment<T>::CopySegment [](start = 30, length = 34)

CopySegment doesn't always allocate a new segment, can we assert before clearing that current != oldSegment